### PR TITLE
feat: connection pooling

### PR DESCRIPTION
**Pull request by Arqu**

### CODE REVIEW

File: iroh-rpc-client/src/client.rs [hunk 1]
```diff
@@ -70,10 +123,11 @@ impl Client {
             .ok_or_else(|| anyhow!("missing rpc gateway connnection"))
     }
 
-    pub fn try_store(&self) -> Result<&StoreClient> {
-        self.store
-            .as_ref()
-            .ok_or_else(|| anyhow!("missing rpc store connnection"))
+    pub fn try_store(&self) -> Result<StoreClient> {
+        if let Some(store_client) = self.store.get() {
```
Comment (rklaehn) [inline iroh-rpc-client/src/client.rs:127]: Same here:
```rust
        self.gateway
            .as_ref()
            .context("missing rpc gateway connnection")
```

**Review by rklaehn — APPROVED** [review]
Looks good. See 2 tiny nits in the comments.

I am hesitant to add so much complexity for inter process communication between processes on the same machine. I think in the longer term we have to redesign the API so that this is not needed. But the code looks good, so it might be a good bandaid for now.

**Review by rklaehn — APPROVED** [review]
Still looks fine as far as the code is concerned

File: iroh-rpc-client/src/client.rs [hunk 2]
```diff
@@ -10,8 +13,56 @@ use crate::store::StoreClient;
...
 pub struct Client {
     pub gateway: Option<GatewayClient>,
-    pub p2p: Option<P2pClient>,
-    pub store: Option<StoreClient>,
+    p2p: P2pLBClient,
+    store: StoreLBClient,
+}
+
+#[derive(Debug, Clone)]
+pub struct StoreLBClient {
```
Comment (dignifiedquire) [inline iroh-rpc-client/src/client.rs:21]: I can haz doc comment of what this does?

File: iroh-rpc-client/src/client.rs [hunk 3]
```diff
@@ -10,8 +13,56 @@ use crate::store::StoreClient;
...
+    clients: Vec<StoreClient>,
+    pos: Arc<AtomicUsize>,
+}
+
+impl StoreLBClient {
+    pub fn get(&self) -> Option<StoreClient> {
```
Comment (dignifiedquire) [inline iroh-rpc-client/src/client.rs:27]: maybe add a comment that this does round robin

**Review by dignifiedquire — DISMISSED** [review]

**Review by b5 — APPROVED** [review]

**Pull request merged**